### DISCUSSION TITLE
Filament forms: Selecting the two-level reactive radio buttons opens a new tab in the browser.

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#2444] [Help]
1) Selecting Answered
![call-answered](https://user-images.githubusercontent.com/29732308/168759961-6e28c896-7477-47c0-a55f-d1e556691eea.png)
2) Selecting Interested and should be open Call Arrangement field
![call-answered-interested](https://user-images.githubusercontent.com/29732308/168759989-7f55c992-1094-409b-84db-cecc9f135a78.png)
3) Select Not answered and then again select answered (all works correctly). Again select not answered and a new tab appears.
![call-answered-new-tab](https://user-images.githubusercontent.com/29732308/168760028-ee3f2fa7-505e-4c66-a37e-bdf3aa67896e.png)

Selecting the two-level reactive radio buttons opens a new tab in the browser.

Selecting `answerStatus` should open the `status` radio buttons.

If `CallStatus::Interested` is selected from the `statuses`, `callArrangementAt` Datetimepicker should be opened.

Everything works as it should, but when I select the `answerStatus` radio button, a new browser tab opens.

```php
protected function getFormSchema(): array
{
    return [
        Radio::make('answerStatus')
                ->options([
                    CallStatus::Answered => __('Answered'),
                    CallStatus::NotAnswered => __('Not answered'),
                ])
                ->reactive()
                ->inline(),

            Radio::make('status')
                ->options([
                    CallStatus::Agreed => __('Agreed'),
                    CallStatus::Interested => __('Interested'),
                    CallStatus::Rejected => __('Rejected'),
                ])
                ->descriptions([
                    CallStatus::Agreed => __('If the carrier has agreed'),
                    CallStatus::Interested => __('If the carrier is interested and you have an appointment to call'),
                    CallStatus::Rejected => __('If the carrier has refused'),
                ])
                ->reactive()
                ->hidden(fn(Closure $get): bool => $get('answerStatus') != CallStatus::Answered())
                ->required(fn(Closure $get): bool => $get('answerStatus') != CallStatus::Answered())
                ->inline(false),

            DateTimePicker::make('callArrangementAt')
                ->hidden(fn(Closure $get): bool => $get('status') != CallStatus::Interested())
                ->required(fn(Closure $get): bool => $get('status') == CallStatus::Interested()),

            Textarea::make('comment')
    ];
}

**Comment by danharrin** [comment]
Could you try again in another browser?

**Reply by ArtMin96** [reply]
> Could you try again in another browser?

There was no such problem in Firefox but it gave another problem in Firefox.
![call-answered-firefox-bug](https://user-images.githubusercontent.com/29732308/168779384-5c0ba4d0-d43e-4289-be53-0108188497e8.png)

**Reply by ArtMin96** [reply]
Fixed.

```php
Radio::make('answerStatus')
    ->options([
        CallStatus::Answered => __('Answered'),
        CallStatus::NotAnswered => __('Not answered'),
    ])
    ->afterStateUpdated(function (Closure $set, Closure $get, $state) {
        if ($get('answerStatus') == CallStatus::NotAnswered()) {
            $set('status', '');
        }
    })
    ->reactive()
    ->inline(),
```

**Reply by danharrin** [reply]
You can replace `$get('answerStatus')` with `$state`

**Reply by ArtMin96** [reply]
Thanks
